packages/central-server/src/hooks/entityAttribute.js

```diff
@@ -0,0 +1,20 @@
+const snakeCased = name =>
+  name
+    .split(/(?=[A-Z])/)
+    .join('_')
+    .toLowerCase();
+
+export async function entityAttribute({ answer, surveyResponse, hookName }) {
+  const entity = await surveyResponse.entity();
+
+  if (!entity) {
+    throw new Error('Invalid entity');
+  }
+
+  const attributeKey = snakeCased(hookName.substring(15));
```

Comment:
When there's a number like this that is defined without context of where it comes from, we call it a "magic number"...and it's a bit of an anti-pattern. Instead much better to show the reader exactly why it's 15 characters long :-)
```suggestion
  const attributeKey = snakeCased(hookName.substring(ENTITY_ATTRIBUTE_HOOK_PREFIX.length));
```

Reply:
Great teachings, @edmofro cheers.